ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzer.java

```diff
@@ -13455,6 +13469,18 @@ ASTNode analyzeCreateTable(
       }
     }
 
+    if (partitionTransformSpecExists) {
+      try {
+        HiveStorageHandler storageHandler = HiveUtils.getStorageHandler(conf, storageFormat.getStorageHandler());
```

Comment:
Can we make this check before, at line 13409? That way we could skip parsing the partition spec unnecessarily first

Reply:
Ah I see. Thanks.